### ISSUE TITLE
hafs_regional_atm_wav fails in debug mode

### ISSUE THREAD

**Issue by DeniseWorthen** [ufs-community/ufs-weather-model#1395]
## Description
Compiling the ``hafs_regional_atm_wav`` test with debug-on results in a seg fault in the cubed-sphere.

### To Reproduce:

On heral.intel, compile and run the test in debug mode.

```
COMPILE | -DAPP=HAFSW -DMOVING_NEST=ON -DCCPP_SUITES=FV3_HAFS_v0_gfdlmp_tedmf,FV3_HAFS_v0_gfdlmp_tedmf_nonsst,FV3_HAFS_v0_thompson_tedmf_gfdlsf -D32BIT=ON -DDEBUG=ON |                | fv3 |
RUN     | hafs_regional_atm_wav                                                                                                   |                                         | fv3 |
```

## Output

The run fails with

```
39: [h2c01:149518:0:149518] Caught signal 8 (Floating point exception: floating-point invalid operation)
  0: [h2c01:149479:0:149479] Caught signal 8 (Floating point exception: floating-point invalid operation)
 19: [h2c01:149498:0:149498] Caught signal 8 (Floating point exception: floating-point invalid operation)
 59: ==== backtrace (tid: 138138) ====
 59:  0 0x000000000004d455 ucs_debug_print_backtrace()  ???:0
 59:  1 0x00000000076fa1df fv_regional_mod_mp_bc_time_interpolation_()  /scratch1/NCEPDEV/nems/Denise.Worthen/WORK/ufs-weather-model/FV3/atmos_cubed_sphere/model/fv_regional_bc.F90:4930
 59:  2 0x00000000076e46e0 fv_regional_mod_mp_regional_boundary_update_()  /scratch1/NCEPDEV/nems/Denise.Worthen/WORK/ufs-weather-model/FV3/atmos_cubed_sphere/model/fv_regional_bc.F90:4615
 59:  3 0x0000000006e795d3 dyn_core_mod_mp_dyn_core_()  /scratch1/NCEPDEV/nems/Denise.Worthen/WORK/ufs-weather-model/FV3/atmos_cubed_sphere/model/dyn_core.F90:566
 59:  4 0x0000000006ffd15c fv_dynamics_mod_mp_fv_dynamics_()  /scratch1/NCEPDEV/nems/Denise.Worthen/WORK/ufs-weather-model/FV3/atmos_cubed_sphere/model/fv_dynamics.F90:679
 59:  5 0x0000000006c54038 atmosphere_mod_mp_adiabatic_init_()  /scratch1/NCEPDEV/nems/Denise.Worthen/WORK/ufs-weather-model/FV3/atmos_cubed_sphere/driver/fvGFS/atmosphere.F90:1922
 59:  6 0x0000000006b7bd59 atmosphere_mod_mp_atmosphere_init_()  /scratch1/NCEPDEV/nems/Denise.Worthen/WORK/ufs-weather-model/FV3/atmos_cubed_sphere/driver/fvGFS/atmosphere.F90:573
 59:  7 0x0000000005e1ad87 atmos_model_mod_mp_atmos_model_init_()  /scratch1/NCEPDEV/nems/Denise.Worthen/WORK/ufs-weather-model/FV3/atmos_model.F90:553
```

**Comment by BinLiu-NOAA** [comment]
@DeniseWorthen, first a clarification is that we did not yet turn on building HAFS application executable in debug mode in the develop branch, correct?
@BijuThomas-NOAA, Could you please help to explore and look into this potential issue. Thanks!

**Comment by DeniseWorthen** [comment]
@BinLiu-NOAA Yes, we do not run any of the hafs tests in debug mode currently. I know there is an issue w/ HYCOM for example. I was doing some ww3 development work and wanted to run this app in debug mode. The issue appears unrelated to the ww3 side however.

**Comment by BinLiu-NOAA** [comment]
@DeniseWorthen, and yes, @BijuThomas-NOAA is working on preparing that HAFS debug RT. And I think he already had a version working, I will let him chime-in for more details.

**Comment by DeniseWorthen** [comment]
@BinLiu-NOAA I believe this issue is now resolved, perhaps through PR #1437. I was able to compile and run the ``hafs_regional_atm_wav`` in debug mode using the latest develop.
